packages/chat-stateful-client/src/ChatContext.ts

```diff
@@ -77,7 +77,7 @@ export class ChatContext {
   }
 
   public createThreadIfNotExist(threadId: string, properties?: ChatThreadProperties): boolean {
-    const exists = this.getState().threads.has(threadId);
+    const exists = Object.prototype.hasOwnProperty.call(this.getState().threads, threadId);
```

Comment:
Curious, is this not equivalent to
`const exists = this.getState().threads[threadId]`

Reply:
The reason is a bit pedantic:

```
const a = {a: 1, b: undefined};

console.log('b', a['b'], Object.prototype.hasOwnProperty.call(a, 'b'));
console.log('c', a['c'], Object.prototype.hasOwnProperty.call(a, 'c'));
```

Gives:

    b undefined true
    c undefined false